pkg/controller/tidb_control.go

```diff
@@ -51,22 +54,40 @@ type TiDBControlInterface interface {
 // defaultTiDBControl is default implementation of TiDBControlInterface.
 type defaultTiDBControl struct {
 	httpClient *http.Client
+	rootCAs    *x509.CertPool
 }
 
 // NewDefaultTiDBControl returns a defaultTiDBControl instance
 func NewDefaultTiDBControl() TiDBControlInterface {
 	httpClient := &http.Client{Timeout: timeout}
-	return &defaultTiDBControl{httpClient: httpClient}
+
+	rootCAs, err := httputil.ReadCACerts()
+	if err != nil {
+		glog.Errorf("fail to load CA certs, use plain connections only, TLS enabled clusters may not work")
+	}
+
+	return &defaultTiDBControl{httpClient: httpClient, rootCAs: rootCAs}
+}
```

Comment:
This func  `NewDefaultTiDBControl ` should return `error` or put the logic to get the rootCA outside of this function.